### PR TITLE
#838 Update/fix the documentation for BITCOUNT command

### PR DESCRIPTION
**Pull request by vaibspider**
- Reformatted the documentation
- Validated all the examples and fixed an incorrect output value
- Added missing parameter details to make it consistent with the code

### CODE REVIEW

**Comment by vaibspider** [comment]
issue #838

File: docs/src/content/docs/commands/BITCOUNT.md [hunk 1]
```diff
@@ -7,115 +7,130 @@ The `BITCOUNT` command in DiceDB is used to count the number of set bits (i.e.,
 
 ## Syntax
 
-```plaintext
-BITCOUNT key [start end]
 ```
+BITCOUNT key [start end [BYTE | BIT]]
```
Comment (apoorvyadav1111) [inline docs/src/content/docs/commands/BITCOUNT.md:11]: Please add a small description for the command.

File: docs/src/content/docs/commands/BITCOUNT.md [hunk 2]
```diff
@@ -7,115 +7,130 @@ The `BITCOUNT` command in DiceDB is used to count the number of set bits (i.e.,
...
+   127.0.0.1:7379> SET mykey "foobar"
+   OK
+   127.0.0.1:7379> BITCOUNT mykey 5 6
+   (integer) 4
+   127.0.0.1:7379> BITCOUNT mykey 5 10
+   (integer) 4
    ```
 
-4. `Out of Range Indices`: If the `start` or `end` parameters are out of the range of the string length, DiceDB will handle it gracefully by considering the valid range within the string.
+## Examples
```
Comment (apoorvyadav1111) [inline docs/src/content/docs/commands/BITCOUNT.md:107]: nit:
```suggestion
## Example Usage
```

File: docs/src/content/docs/commands/BITCOUNT.md [hunk 3]
```diff
@@ -7,115 +7,130 @@ The `BITCOUNT` command in DiceDB is used to count the number of set bits (i.e.,
...
-   BITCOUNT mykey 10 20
-   ```
+### Example 1: Counting bits in the entire string
 
-   `Output:`
+```bash
+127.0.0.1:7379> SET mykey "foobar"
+OK
+127.0.0.1:7379> BITCOUNT mykey
+26
```
Comment (apoorvyadav1111) [inline docs/src/content/docs/commands/BITCOUNT.md:115]:
```suggestion
(integer) 26
```

File: docs/src/content/docs/commands/BITCOUNT.md [hunk 4]
```diff
@@ -7,115 +7,130 @@ The `BITCOUNT` command in DiceDB is used to count the number of set bits (i.e.,
...
-   0
-   ```
+### Example 2: Counting bits in a specified range
 
-## Notes
+```bash
+127.0.0.1:7379> SET mykey "foobar"
+OK
+127.0.0.1:7379> BITCOUNT mykey 1 3
+15
```
Comment (apoorvyadav1111) [inline docs/src/content/docs/commands/BITCOUNT.md:124]:
```suggestion
(integer) 15
```

File: docs/src/content/docs/commands/BITCOUNT.md [hunk 5]
```diff
@@ -7,115 +7,130 @@ The `BITCOUNT` command in DiceDB is used to count the number of set bits (i.e.,
...
 
-- The `start` and `end` parameters are byte positions, not bit positions.
-- Negative values for `start` and `end` are interpreted as offsets from the end of the string. For example, -1 means the last byte, -2 means the second last byte, and so on.
+### Example 3: Counting bits in a specified range using bit indices
+
+```bash
+127.0.0.1:7379> SET mykey "foobar"
+OK
+127.0.0.1:7379> BITCOUNT mykey 8 31 BIT
+15
```
Comment (apoorvyadav1111) [inline docs/src/content/docs/commands/BITCOUNT.md:134]:
```suggestion
(integer) 15
```

**Review by apoorvyadav1111 — CHANGES_REQUESTED** [review]
Hi, thanks for the changes. have left some comments to ensure the documentation is consistent with other commands.

File: docs/src/content/docs/commands/BITCOUNT.md (hunk 1 above)
Comment (vaibspider) [reply docs/src/content/docs/commands/BITCOUNT.md:11]: I think it's already added at line no. 6. Do you want me to reshuffle the order or duplicate the description?
Just wanted to confirm, thanks!

File: docs/src/content/docs/commands/BITCOUNT.md (hunk 2 above)
Comment (vaibspider) [reply docs/src/content/docs/commands/BITCOUNT.md:107]: I think in the original issue for this task/PR mentions the title to be "Examples": [https://github.com/DiceDB/dice/issues/838](https://github.com/DiceDB/dice/issues/838).

> Note: The title of the sections should be the exact strings mentioned above, like "Examples", "Return values", etc.

So can you confirm which one should we adhere to? Thanks!

File: docs/src/content/docs/commands/BITCOUNT.md (hunk 5 above)
Comment (vaibspider) [reply docs/src/content/docs/commands/BITCOUNT.md]: Done!

File: docs/src/content/docs/commands/BITCOUNT.md (hunk 4 above)
Comment (vaibspider) [reply docs/src/content/docs/commands/BITCOUNT.md]: Done!

File: docs/src/content/docs/commands/BITCOUNT.md (hunk 3 above)
Comment (vaibspider) [reply docs/src/content/docs/commands/BITCOUNT.md]: Done!

File: docs/src/content/docs/commands/BITCOUNT.md (hunk 2 above)
Comment (apoorvyadav1111) [reply docs/src/content/docs/commands/BITCOUNT.md:107]: Please take a look at this as an example. https://dicedb.io/commands/set/. I understand there is inconsistency but let's try to make it similar to other pages. Thanks.

Comment (vaibspider) [reply docs/src/content/docs/commands/BITCOUNT.md]: Ok, done!

**Comment by apoorvyadav1111** [comment]
Hi @vaibspider, thanks for making the changes. Please confirm if it's ready to be merged.

**Comment by vaibspider** [comment]
@apoorvyadav1111 -- Yes, it's ready to be merged. I've done the changes you suggested.
Thanks for reviewing!

**Pull request merged**